src/NeoSharp.BinarySerialization/Extensions/BinaryReaderExtensions.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Linq;
+using System.Text;
+
+namespace System.IO
+{
+    public static class BinaryReaderExtensions
+    {
+        public static byte[] ReadBytesWithGrouping(this BinaryReader reader)
+        {
+            const int groupSize = 16;
+            using (var ms = new MemoryStream())
+            {
+                int padding;
+                do
+                {
+                    var group = reader.ReadBytes(groupSize);
+                    padding = reader.ReadByte();
+                    if (padding > groupSize)
+                        throw new FormatException();
+                    var count = groupSize - padding;
+                    if (count > 0)
+                        ms.Write(group, 0, count);
+                } while (padding == 0);
+                return ms.ToArray();
+            }
+        }
+
+        public static string ReadFixedString(this BinaryReader reader, int length)
```

Comment:
BinaryReader is capable to read fixed string if it was serialized before with length + payload. Why do we need this method?